scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -1103,25 +1113,43 @@ class Router(formatOps: FormatOps) {
         val onlyConfigStyle =
           mustUseConfigStyle(formatToken, !opensLiteralArgumentList)
 
+        val argsOpt = (leftOwner match {
+          case SplitCallIntoParts(_, args) =>
+            args.fold(Some(_), formatOps.findArgsFor(open, _))
+          case _ => None
+        }).filter(_.nonEmpty)
+        val isSingleArg = argsOpt.exists(_.lengthCompare(1) == 0)
+
+        def findComma(ft: FormatToken) =
```

Comment:
Should we extract it? It's also used above. The logic overall seems very similar.

Reply:
Also this seems to be a very similar logic repeated 3 times, would it be possible reduce at least some parts of it?

Reply:
done